src/board/generic_esp32/generic-esp32-board.c

```diff
@@ -179,8 +282,12 @@ void RtcBkupRead( uint32_t *data0, uint32_t *data1 )
  */
 uint32_t RtcGetCalendarTime( uint16_t *milliseconds )
 {
-    *milliseconds = 29;
-    return  42 /* seconds */;
+    int64_t time_us = esp_timer_get_time();
+    uint32_t time_s = (uint32_t)(time_us / US_PER_S);
+
+    *milliseconds = (uint16_t)(time_us * US_PER_MS);
```

Comment:
this doesn't look correct, milliseconds appears to be the remainder of time since epoch when you've taken away all of the whole seconds.